### PR TITLE
Routing branch v3

### PR DESCRIPTION
**Pull request by OzLustig**
Dealt with the basic structure of the page and created basic Angular routing in between the different components.

### CODE REVIEW

File: projects/program-website/src/app/app.module.ts [hunk 1]
```diff
@@ -4,15 +4,26 @@ import { NgModule } from '@angular/core';
 import { AppComponent } from './app.component';
 import { CoreModule } from './core/core.module';
 import { FirebaseModule } from './import/firebase/firebase.module';
+import { AboutComponent } from './about-component/about.component';
+import { ProjectsComponent } from './projects-component/projects.component';
+import { ParticipantsComponent } from './participants-component/participants.component';
+import { ContactUsComponent } from './contact-us-component/contact-us.component';
+import {RouterModule} from '@angular/router';
+import { AppRoutingModule } from './app-routing.module';
```
Comment (webdevelopland) [inline projects/program-website/src/app/app.module.ts]: It's recommended to separate third-party libraries imports and imports from the app:
https://angular.io/guide/styleguide#import-line-spacing

File: projects/program-website/src/app/core/services/firebase.service.ts~HEAD [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
```
Comment (webdevelopland) [inline projects/program-website/src/app/core/services/firebase.service.ts~HEAD]: You probably added the file by accident.
Could you please remove it?

File: projects/program-website/src/app/projects-component/projects.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (webdevelopland) [inline projects/program-website/src/app/projects-component/projects.component.spec.ts]: I'm not sure you know it, but the unit test actually does nothing. It contain some many cryptic code and it's useless.
My opinion, we shouldn't keep it, unless you add some real tests here.
Even if the test is created automatically by `ng`.

File: projects/program-website/package.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
     "@types/node": "~8.9.4",
     "angularfire2": "^5.1.1",
     "codelyzer": "~4.5.0",
-    "firebase": "^5.8.1",
+    "firebase": "^5.7.3",
```
Comment (webdevelopland) [inline projects/program-website/package.json]: :slightly_smiling_face:

File: projects/program-website/src/app/app.component.html [hunk 5]
```diff
@@ -1,6 +1,17 @@
 <div *ngIf="!book">
   Loading
 </div>
+<nav class="pull-right">
+  <a routerLink="/projects">  Projects  </a>
+  <a routerLink="/participants">  Participants  </a>
+  <a routerLink="/about"> About </a>
```
Comment (webdevelopland) [inline projects/program-website/src/app/app.component.html]: Why did you add space here? `Projects` has even 2 spaces.
FYI: html ignore more than 1 space.

Why not:
```html
<a routerLink="/projects">Projects</a>
```
If you tried to separate menu options, then it's be better to use css for the purpose.

File: projects/program-website/src/app/app-routing.module.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import {ProjectsComponent} from '@/projects-component/projects.component';
+import {AboutComponent} from '@/about-component/about.component';
+import {ContactUsComponent} from '@/contact-us-component/contact-us.component';
+import {ParticipantsComponent} from '@/participants-component/participants.component';
```
Comment (webdevelopland) [inline projects/program-website/src/app/app-routing.module.ts]: BTW, could you use constant format style in your code?
If first 2 imports has space between `{}`, then why other don't?
You can fix it automatically by code formatter in your code editor.
VS code: Shift+Ctrl+I
Sublime: Ctrl+T+F

File: projects/program-website/src/app/app.module.ts (hunk 1 above)
Comment (OzLustig) [reply projects/program-website/src/app/app.module.ts]: Handled.

import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import {RouterModule} from '@angular/router';

import { AppComponent } from './app.component';
import { CoreModule } from './core/core.module';
import { FirebaseModule } from './import/firebase/firebase.module';
import { AboutComponent } from './about-component/about.component';
import { ProjectsComponent } from './projects-component/projects.component';
import { ParticipantsComponent } from './participants-component/participants.component';
import { ContactUsComponent } from './contact-us-component/contact-us.component';
import { AppRoutingModule } from './app-routing.module';

File: projects/program-website/src/app/core/services/firebase.service.ts~HEAD (hunk 2 above)
Comment (OzLustig) [reply projects/program-website/src/app/core/services/firebase.service.ts~HEAD]: Handled.

File: projects/program-website/src/app/app.component.html (hunk 5 above)
Comment (OzLustig) [reply projects/program-website/src/app/app.component.html]: Handled.

File: projects/program-website/src/app/app.component.html [hunk 7]
```diff
@@ -2,10 +2,10 @@
...
 </div>
 <nav class="pull-right">
-  <a routerLink="/projects">  Projects  </a>
-  <a routerLink="/participants">  Participants  </a>
-  <a routerLink="/about"> About </a>
-  <a routerLink="/contact"> Contact Us! </a>
+  <a routerLink="/projects">Projects</a>&nbsp;
+  <a routerLink="/participants">Participants</a>&nbsp;
+  <a routerLink="/about">About</a>&nbsp;
+  <a routerLink="/contact">Contact Us</a>&nbsp;
```
Comment (webdevelopland) [inline projects/program-website/src/app/app.component.html]: Interesting solution, but usually developers don't add html special chars without necessity.
To add space between options you could add css style to `a` element
```scss
.pull-right {
  a {
    margin-right: 10px;
  }
}
```

File: projects/program-website/src/app/core/services/firebase.service.ts~HEAD [hunk 8]
```diff
@@ -1,4 +1,3 @@
-import { Injectable } from '@angular/core';
```
Comment (webdevelopland) [inline projects/program-website/src/app/core/services/firebase.service.ts~HEAD]: Not just first string. Whole file.
Look at the name:
```
firebase.service.ts~HEAD
```
Also, we already have firebase service.
So, can you delete the file?

File: projects/program-website/src/app/about-component/about.component.spec.ts [hunk 9]
```diff
@@ -1,25 +0,0 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (webdevelopland) [inline projects/program-website/src/app/about-component/about.component.spec.ts]: Thanks for deleting it. But why not to delete the files too?
